week-3/mandatory/2-api/cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,214 @@
+    db.query(sql, [id],
+        (err, result) => {
+            err ? res.status(500).send(err)
+                : res.json(result.rows[0])
+        })
+})
+
+app.get('/customers/:id', (req, res) => {
+    const id = req.params.id
+    db.query('select * from customers where id=$1', [id],
+        (err, result) => {
+            !err && res.json(result.rows[0])
+
+        })
+})
+
+app.get('/suppliers', (req, res) => {
+    const searchName = req.query.name
+    const searchCountry = req.query.country
+
+    const attributes = [searchName, searchCountry]
+    const sql = 'select * from suppliers where (supplier_name = $1 or ($1 is null and 1=1)) or (country = $2 or ($2 is null and 1=1));';
+
+    attributes.some(i => i != undefined) ? db.query(sql, attributes,
+        (err, result) => {
+            res.json(result.rows)
+        }) : db.query('select * from suppliers',
+            (err, result) => {
+                res.json(result.rows)
+            })
+});
+
+
+app.post('/customers', (req, res) => {
+    const name = req.body.name;
+    const city = req.body.city;
+    const address = req.body.address;
+    const country = req.body.country;
+
+    const sql = "insert into customers (name, city, address, country) values ($1, $2, $3, $4) returning id"
+    const attributes = [name, city, address, country]
+    db.query(sql, attributes, (err, result) => {
+        err ? res.status(500).send(err)
+            : res.json(`New customer created with ${result.rows[0].id}`);
+    })
+})
+
+app.post('/products', (req, res) => {
+    const name = req.body.product_name;
+    const sql = "insert into products (product_name) values ($1) returning id"
+    db.query(sql, [name], (err, result) => {
+        err ? res.status(500).send(err)
+            : res.json(`New product created with ${result.rows[0].id}`);
+    })
+})
+
+app.post('/availability', (req, res) => {
+    const prod_id = req.body.prod_id;
+    const supp_id = req.body.supp_id;
+    const price = req.body.unit_price;
+    if (price < 0) {
+        res.status(404).send('Price hast to be greter than 0');
+        return
+    }
+    const sql_prod = "select * from product_availability where prod_id = $1"
+    db.query(sql_prod, [prod_id], (err, result) => {
+        if (err || result.rows.length == 0) {
+            res.status(500).send(err, "Didnt match foreign key prod_id")
+            return;
+        }
+    })
+    const sql_supp = "select * from product_availability where supp_id = $1"
+    db.query(sql_supp, [supp_id], (err, result) => {
+        if (err || result.rows.length == 0) {
+            res.status(500).send(err, "Didnt match foreign key supp_id")
```

Comment:
a much friendlier message  is usually sent to the user, you never send db secrets